greenplumpython/experimental/file.py

```diff
@@ -0,0 +1,79 @@
+import base64
+import inspect
+import io
+import pathlib
+import tarfile
+import uuid
+from typing import get_type_hints
+
+import psycopg2
+
+import greenplumpython as gp
+from greenplumpython.func import NormalFunction
+
+_CHUNK_SIZE = 256 * 1024 * 1024  # Must be much < 1 GB
+
+
+@gp.create_function
+def _dump_file_chunk(tmp_archive_name: str, chunk_base64: str) -> int:
+    tmp_archive_base = pathlib.Path("/") / "tmp" / tmp_archive_name
+    tmp_archive_path = tmp_archive_base / f"{tmp_archive_name}.tar.gz"
+    with open(tmp_archive_path, "ab") as tmp_archive:
+        tmp_archive.write(base64.b64decode(chunk_base64))
+    return 0
+
+
+@gp.create_function
+def _extract_files(tmp_archive_name: str) -> list[str]:
+    tmp_archive_base = pathlib.Path("/") / "tmp" / tmp_archive_name
```

Comment:
An alternative solution could be something like:

```
COPY base64decode(xxx) to 'cat >> /tmp/xxx.tar.gz`
```

The current one is OK.

Reply:
Uh, OK. `COPY` has a `NEWLINE` option, but it needs one of `CR`/`LF`/`CRLF`. 
This may not work well without further hacks.